test/test_images/probe_helper/probe_helper_test.go

```diff
@@ -236,48 +236,35 @@ func testStorageClient(ctx context.Context, t *testing.T) (*storage.Client, chan
 	return c, gotRequest, srv.Close
 }
 
-func testBrokerClient(ctx context.Context, t *testing.T) (*storage.Client, chan *http.Request, func()) {
-	gotRequest := make(chan *http.Request, 1)
-	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// The test Cloud Storage server forwards the client's generated HTTP requests.
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
-			logging.FromContext(ctx).Fatal("Test Cloud Storage server could not read request body.")
-		}
-		r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
-		gotRequest <- r
-		w.Write([]byte("{}"))
-	}))
-	c, err := storage.NewClient(ctx, option.WithEndpoint(srv.URL))
-	if err != nil {
-		t.Fatalf("Failed to create test storage client: %v", err)
-	}
-	return c, gotRequest, srv.Close
-}
-
 type eventAndResult struct {
 	event      *cloudevents.Event
 	wantResult protocol.Result
 }
 
 func TestProbeHelper(t *testing.T) {
-	t.Skip("Skip this test from running on Prow as it is only for local development.")
-
-	os.Setenv("K_SINK", testBrokerURL)
-	os.Setenv("PROJECT_ID", testProjectID)
-	os.Setenv("CLOUDPUBSUBSOURCE_TOPIC_ID", testTopicID)
-	os.Setenv("CLOUDSTORAGESOURCE_BUCKET_ID", testStorageBucket)
-
 	ctx := logtest.TestContextWithLogger(t)
 	ctx = WithProjectKey(ctx, testProjectID)
 	ctx = WithTopicKey(ctx, testTopicID)
 	ctx = WithSubscriptionKey(ctx, testSubscriptionID)
+	ctx = cloudevents.ContextWithRetriesConstantBackoff(ctx, 100*time.Millisecond, 20)
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	// Set up ports for testing the probe helper.
+	receiverPort, err := GetFreePort()
+	if err != nil {
+		t.Fatalf("Failed to get free receiver port: %v", err)
+	}
+	receiverURL := fmt.Sprintf("http://localhost:%d", receiverPort)
+	probePort, err := GetFreePort()
```

Comment:
Can this call return the same port as the previous one?

Reply:
It never seems to happen, the port assignment is not random and tends to be incremental (+1 from the last one assigned).

Reply:
The port is closed in GetFreePort because otherwise it would already be in use once the client would attempt to open it. If we want to use the port without closing it, we should pass a listener to each client in the tests, and allow for passing listeners to the probe helper, which is what I did before.

I've been combing through the port assignment code but I haven't been able to find the assignment strategy, experiments definitely suggest that GetFreePort will not return the same port twice in a row though. Up to you if I should go back to using listeners or stick with it this way.

Reply:
Thanks for running the experiment! I am fine with it now. We can deal with it later if it bites us, but seems very unlikely